aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -380,8 +386,56 @@ static void copy_kernel_cuda(TensorIterator& iter, bool non_blocking) {
     // TODO: warn on the return value.
     CachingHostAllocator_recordEvent(ptr, ctx, stream);
 
-  } else {
-    at::cuda::memcpy_and_sync(dst, src, nbytes, kind, stream);
+  } else { 
+    if (iter.ndim() <= 2 && !iter.is_contiguous()) {       
```

Comment:
nit maybe move this logic to a shared function with the check above to ensure it stays in sync